android/src/main/java/com/lwansbrough/RCTCamera/RCTCameraModule.java

```diff
@@ -175,6 +194,185 @@ public String getName() {
+            mMediaRecorder.setMaxDuration(totalSeconds * 1000);
+        }
+
+        if (options.hasKey("maxFileSize")) {
+            int maxFileSize = options.getInt("maxFileSize");
+            mMediaRecorder.setMaxFileSize(maxFileSize);
+        }
+
+        try {
+            mMediaRecorder.prepare();
+        } catch (IllegalStateException e) {
+            releaseMediaRecorder();
+            return false;
+        } catch (IOException e) {
+            releaseMediaRecorder();
+            return false;
+        }
+        return true;
+
+    }
+
+    @ReactMethod
+    private void record(final ReadableMap options, final Promise promise) {
+        if (mRecordingPromise == null) {
+            mCamera = RCTCamera.getInstance().acquireCameraInstance(options.getInt("type"));
+            if (null == mCamera) {
+                promise.reject("No camera found.");
+                return;
+            }
+            if (!prepareMediaRecorder(options)) {
+                promise.reject("Fail in prepareMediaRecorder()!");
+                return;
+            }
+            try {
+                mMediaRecorder.start();
+                MRStartTime =  System.currentTimeMillis();
+                mRecordingOptions = options;
+                mRecordingPromise = promise;  // only got here if mediaRecorder started
+            } catch (final Exception ex) {
+                promise.reject("Exception in thread");
+                return;
+            }
+        }
+    }
+
+    private void releaseMediaRecorder() {
+        if (mRecordingPromise != null) {
+            // Must record at least a second or MediaRecorder throws exceptions on some platforms
+            long duration = System.currentTimeMillis() - MRStartTime;
+            if (duration < 1500) {
+                try {
+                    Thread.sleep(1500-duration);
+                } catch(InterruptedException e) {
+                    e.printStackTrace();
+                }
+            }
+            try {
+                mMediaRecorder.stop(); // stop the recording
+            } catch (RuntimeException e) {
+                e.printStackTrace();
+            }
+        }
+        mMediaRecorder.reset(); // clear recorder configuration
+
+        if (mCamera != null) {
+            mCamera.lock(); // relock camera for later use since we unlocked it
+        }
+
+        // Make sure readable
+        File f = new File(mVideoFile.getPath());
+        if (f.exists()) {
+            f.setReadable(true, false); // so mediaplayer can play it
+            f.setWritable(true, false); // so can clean it up
+
+            if (mRecordingPromise != null) {
+                switch (mRecordingOptions.getInt("target")) {
+                    case RCT_CAMERA_CAPTURE_TARGET_MEMORY:
+                        byte[] encoded = convertFileToByteArray(mVideoFile);
+                        mRecordingPromise.resolve(new String(encoded, Base64.DEFAULT));
```

Comment:
This should be `resolve`d with the new capture return options.

https://github.com/lwansbrough/react-native-camera#captureoptions-promise